packages/Meshfree/src/DTK_DetailsNearestNeighborOperatorImpl.hpp

```diff
@@ -138,11 +147,39 @@ struct NearestNeighborOperatorImpl
             DTK_MARK_REGION( "set_target_values" ),
             Kokkos::RangePolicy<ExecutionSpace>( 0, n_imports ),
             KOKKOS_LAMBDA( int i ) {
-                target_values( import_target_indices( i ) ) =
-                    import_source_values( i );
+                for ( int j = 0; j < (int)target_values.dimension_1(); ++j )
+                    target_values( import_target_indices( i ), j ) =
+                        import_source_values( i, j );
             } );
         Kokkos::fence();
     }
+
+    template <typename View>
+    static typename View::non_const_type
+    fetch( Teuchos::RCP<Teuchos::Comm<int> const> const &comm,
```

Comment:
I am confused on what this function does. Why does fetch does a `pull` and a `push`? `fetch` for me means that it only does a `pull`

Reply:
`pullSourceValues()` and `pushTargetValues()` are implementation details. `fetch()` allow you to retrieve values on a distributed array.